### ISSUE TITLE
ENH: add lzma (.xz) support to io.fits 

### ISSUE THREAD

**Issue by dhomeier** [astropy/astropy#9714]
### Description
Loosely related to #9652 and #9685, is there a specific reason why `io.fits` does not support `lzma` compression? From the way `.xz` files are handled in [data.py](https://github.com/astropy/astropy/blob/274e66d0c626e341695165e4e55c53ec48d31881/astropy/utils/data.py#L302) it looks like it could be implemented quite analogously in the FITS file handlers, but of course it would involve somewhat more significant modifications.

**Comment by saimn** [comment]
Not specific reason I think, it is probably just less frequently used and nobody needed it.

**Comment by ivvv** [comment]
One attractive feature is that the `xz` allows multi-threading compression/decompression, which is useful with huge FITS files.

**Comment by dhomeier** [comment]
Right. Though in most such cases the tiled compression capacity of `CompImageHDU` would probably be the better choice now, linking this to #15474.

**Comment by saimn** [comment]
xz/lzma is available through python's stdlib (https://docs.python.org/3/library/lzma.html) so adding support should be straightforward.

**Comment by dhomeier** [comment]
And we already have an equivalent setup for `HAS_BZ2`

**Comment by kYwzor** [comment]
Came across this while working on #17960. Not sure how common these actually are, but the code necessary for this is so simple I just created a PR. I could find [at least one instance of "fits.xz" files in the wild](https://github.com/liweitianux/fg21sim/blob/master/docs/data.rst), so there are real use-cases (if rare).
